packages/fuels-types/src/function_selector.rs

```diff
@@ -105,3 +133,230 @@ fn build_fn_selector_params(prop: &Property) -> String {
     }
     result
 }
+
+// @todo This is an experimental support for the new JSON ABI file format.
+// Once this is stable:
+// 1. Delete old one;
+// 2. Rename it to its original name;
+// 3. Write documentation.
+fn _new_build_fn_selector_params(
+    prop: &TypeDeclaration,
+    types: &HashMap<usize, TypeDeclaration>,
+) -> String {
+    let mut result: String = String::new();
+
+    let type_declaration = types.get(&prop.type_id).expect("couldn't find type");
+
```

Comment:
This `.unwrap()` will break the macro under undesirable scenarios. If it's not a custom type, `components` might be `None`, which is correct but will cause this section to panic. That's why we only compute the number of elements from within the `type_declaration.is_custom_type(types)` if-block. 


Tip: make sure to run the code locally with a given suggestion, otherwise a click on `Commit suggestion` might break the build!

Reply:
Yeah I messed up on this lol